### ISSUE TITLE
ADB backup mismatching md5

### ISSUE THREAD

**Issue by baltitenger** [omnirom/android_bootable_recovery#249]
- [ ] I am running an official build of TWRP, downloaded from https://twrp.me/Devices/
- [x] I am running the latest version of TWRP
- [x] I have read the FAQ (https://twrp.me/FAQ/)
- [x] I have searched for my issue and it does not already exist

**Device codename**: tissot
**TWRP version**: 3.2.1

#### WHAT STEPS WILL REPRODUCE THE PROBLEM?
```bash
adb backup --twrp boot system data -f test.ab
adb restore test.ab
```

#### WHAT IS THE EXPECTED RESULT?
It should restore the backup.

#### WHAT HAPPENS INSTEAD?
It successfully restores boot, but it fails at the next partition: `E:md5 doesn't match!`, and this happens consistently.

#### ADDITIONAL INFORMATION
There should be an option to skip digest check.

[`/tmp/recovery.log`](https://paste.omnirom.org/view/cc308d27)
[`dmesg`](https://paste.omnirom.org/view/243a2a89)

**Comment by baltitenger** [comment]
Well...
I've checked it now with an official twrp (3.2.3-1), and it seems to be working fine. Now I just don't know how to install this one. My device doesn't have a recovery partition, so I can only hotboot as far as I know.

**Comment by baltitenger** [comment]
Turns out it's supposed to be boot, not recovery. You should change your tutorial for devices like mine.

**Comment by TheOct0** [comment]
I am sorry, but I am facing the same issue, and I would appreciate it if you described the way you solved it in a little more detail. I have this md5 issue with a backup made with twrp 3.2.1, and switching to 3.2.3 only gives me an error about an incompatible adb backup version.

**Comment by baltitenger** [comment]
I couldn't get the backup made with the unofficial twrp working. I had the same issue. By 'working fine' i meant that I'm able to take backups and successfully restore them. As far as I know, that backup is done for. I've also tried extracting data from it, but gunzip also fails with mismatching md5 and I don't see an option to disable md5 check, I guess that just isn't supposed to happen.

**Comment by TheOct0** [comment]
I've been trying editing enc recompiling TWRP in order to make it bypass the MD5 check, but I haven't had much luck (always fail somewhere before actually going on my phone, I'm just bad at compiling). Thankfully, I was able to recover my email address and bypass the 2FA authentication (which was made impossible without my backup). Long story short, doesn't work, probably never will, I'm a sad little boy, but at least I got my work email address back.
